routes/shared.go

```diff
@@ -342,9 +349,92 @@ func (fes *APIServer) putUserMetadataInGlobalState(
 
 	return nil
 }
+func etherToWei(val *big.Int) *big.Int {
+	return new(big.Int).Mul(val, big.NewInt(params.Ether))
+}
+func (fes *APIServer) SendDesoMetamask(recipientPkBytes string) (txnHash *lib.BlockHash, _err error) {
+	// Alright so there's been a lot of trial and error on getting things to work so instead of over
+	// focusing on the small details I figured it would make the
+	// most sense to run a high level code execution plan by you guys instead.
+
+	// 1. validate the request to ensure they aren't trying to drain the account.
+	publicKey, _, err := lib.Base58CheckDecode(recipientPkBytes)
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		return nil, err
+	}
+	balance, err := utxoView.GetDeSoBalanceNanosForPublicKey(publicKey)
+	if err != nil {
+		return nil, err
+	}
+	if balance > 0 {
+		err := errors.New("account already has Deso")
+		return nil, err
+	}
```

Comment:
We probably need to add an entry to the global state that says you received the money in the past. Imagine you get the starter money and then continously send it away, dropping your balance to zero.

Reply:
oh aren't we checking DeSo balance here? And yeah we could either make a global state entry (probably simplest) or check if they executed a transaction in the past -- though getting that view would be kinda tricky. I don't think we have a ready-to-use index for this but could be wrong.